public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts

```diff
@@ -40,15 +39,38 @@ interface TemplateSearchResult {
   promql: string | null;
 }
 
-function getExplainMessage({
-  documentation,
-  metricType,
-  description,
-  query,
-}: ExplainUserPromptParams): llms.openai.Message[] {
+export function getExplainMessage(
+  query: string,
+  metric: string,
+  datasource: PrometheusDatasource,
+): llms.openai.Message[] {
+
+  let metricMetadata = '';
+  let metricType = '';
+
+  const pvq = buildVisualQueryFromString(query);
+
+  if (datasource.languageProvider.metricsMetadata) {
+    metricType = getMetadataType(metric, datasource.languageProvider.metricsMetadata!) ?? '';
+    metricMetadata = getMetadataHelp(metric, datasource.languageProvider.metricsMetadata!) ?? '';
+  }
+
+  const documentationBody = pvq.query.operations.map((op) => {
+    const def = promQueryModeller.getOperationDef(op.id);
+    if (!def) {
+      return '';
+    }
+    const title = def.renderer(op, def, '<expr>');
+    const body = def.explainHandler ? def.explainHandler(op, def) : def.documentation ?? 'no docs';
```

Comment:
Would it be better to not mention docs at all in the prompt, if the docs are missing?